### PR TITLE
Feature: load env automatically

### PR DESCRIPTION
**Pull request by Anna-Xiong**
## Description

**Fearure: when vizro-ai is imported, we load `.env` in the same directory where the program is run.** 

limitation: 
- The load_dotenv function, by default, searches for .env files starting from the current directory and moving upwards, as detailed in the [python-dotenv documentation](https://saurabh-kumar.com/python-dotenv/reference/). This search behavior is typically sufficient for most projects.

-  User can overwrite the default import of .env manually as below:
```
env_file = join(dirname(__file__), '.env')
# or with different name
env_file = find_dotenv(".env.dev")
load_dotenv(env_file)
```

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

**Comment by antonymilne** [comment]
FYI I just checked and it looks like `load_env` will by default search upwards through the directory hierarchy for files named `.env`: https://saurabh-kumar.com/python-dotenv/reference/. So the restriction "it has to be in the same directory" isn't actually true, even though it's likely the most common way people will run this.

Just leaving this with the default `load_env` behaviour looks good to me 👍 And anyone who wants to change that can do the `load_env` manually.

**Comment by Anna-Xiong** [comment]
> FYI I just checked and it looks like `load_env` will by default search upwards through the directory hierarchy for files named `.env`: https://saurabh-kumar.com/python-dotenv/reference/. So the restriction "it has to be in the same directory" isn't actually true, even though it's likely the most common way people will run this.
> 
> Just leaving this with the default `load_env` behaviour looks good to me 👍 And anyone who wants to change that can do the `load_env` manually.

Hi Antony, thank you for pointing this out! yes, i just tested, looks like `find_dotenv` is automatically used. Then we will just add some documentations! 

TL;DR
if we ever want to enable more default file names in future, we can consider similar approach like this one:
 - https://github.com/pallets/flask/blob/7b5e176d1ab23e183d0403573358299ed6562dce/src/flask/cli.py#L656
 
starting from L656 that ths cli.load_dotenv is the function flask uses when app.run() is called inside of run call.

**Comment by antonymilne** [comment]
Agreed 👍 Though note that `.env` is the one that flask recommends be used for private variables and hence probably always makes most sense for API tokens.

> Variables set on the command line are used over those set in .env, which are used over those set in .flaskenv. .flaskenv should be used for public variables, such as FLASK_APP, while .env should not be committed to your repository so that it can set private variables.

File: vizro-ai/changelog.d/20240118_144903_anna_xiong_load_env_automatically.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+<!--
+### Removed
+
+- A bullet item for the Removed category with a link to the relevant PR at the end of your entry, e.g. Enable feature XXX ([#1](https://github.com/mckinsey/vizro/pull/1))
+
+-->
+
+### Added
+
+- Enable feature automatically loading of .env ([#270](https://github.com/mckinsey/vizro/pull/270))
```
Comment (maxschulz-COL) [inline vizro-ai/changelog.d/20240118_144903_anna_xiong_load_env_automatically.md:22]:
```suggestion
- Enable automatic loading of .env ([#270](https://github.com/mckinsey/vizro/pull/270))
```

File: vizro-ai/docs/pages/tutorials/explore_vizro_ai.md [hunk 2]
```diff
@@ -13,12 +13,11 @@ Before proceeding, ensure the installation of the `vizro_ai` package by followin
 
 A prerequisite for this tutorial is access to one of the supported large language models. Please refer to the [api setup](../user_guides/api_setup.md) for instructions on setting up the API.
 
-Upon successful setup, load your API key with the following two lines:
+Upon successful setup, your api key should be ready in the environment when you import `vizro_ai`
```
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/tutorials/explore_vizro_ai.md:16]:
```suggestion
Upon successful setup, your API key should be ready in the environment when you import `vizro_ai`
```

File: vizro-ai/docs/pages/tutorials/explore_vizro_ai.md [hunk 3]
```diff
@@ -13,12 +13,11 @@ Before proceeding, ensure the installation of the `vizro_ai` package by followin
...
 
-```py
-from dotenv import load_dotenv
-load_dotenv()
-```
+If you would like to customize the .env file location and name, you can set it manually.
```
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/tutorials/explore_vizro_ai.md:18]:
```suggestion
If you would like to customize the `.env` file location and name, you can set it manually.
```

File: vizro-ai/docs/pages/tutorials/explore_vizro_ai.md [hunk 4]
```diff
@@ -13,12 +13,11 @@ Before proceeding, ensure the installation of the `vizro_ai` package by followin
...
+You can override the default import of the .env file by specifying the path and name of your custom .env file.
```
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/tutorials/explore_vizro_ai.md:19]:
```suggestion
You can override the default import of the `.env` file by specifying the path and name of your custom `.env` file.
```

File: vizro-ai/docs/pages/tutorials/explore_vizro_ai.md [hunk 5]
```diff
@@ -13,12 +13,11 @@ Before proceeding, ensure the installation of the `vizro_ai` package by followin
...
+Please refer to [api setup](../user_guides/api_setup.md) for instructions on customizing the .env file location and name.
```
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/tutorials/explore_vizro_ai.md:20]:
```suggestion
Please refer to [api setup](../user_guides/api_setup.md) for instructions on customizing the `.env` file location and name.
```

File: vizro-ai/docs/pages/user_guides/api_setup.md [hunk 6]
```diff
@@ -18,16 +18,33 @@ file. Then, you can load the API key from the `.env` file in your development en
 Avoid committing the `.env` file if you are using a version control system such as Git.
 You can add `.env` to your `.gitignore` file to avoid committing it.
 
+By default, when vizro-ai is imported, it automatically loads the `.env` file.
```
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user_guides/api_setup.md:21]:
```suggestion
By default, when `vizro-ai` is imported, it automatically loads the `.env` file.
```

File: vizro-ai/docs/pages/user_guides/api_setup.md [hunk 7]
```diff
@@ -18,16 +18,33 @@ file. Then, you can load the API key from the `.env` file in your development en
...
         ```text
         OPENAI_API_KEY=<your API key>
         ```
-    === "Jupyter Notebook"
-        ```python
-        from dotenv import load_dotenv
-        load_dotenv()
-        ```
+#### Customizing the .env file location and name
+If you would like to customize the .env file location and name, you can set it manually.
```
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user_guides/api_setup.md:30]:
```suggestion
If you would like to customize the `.env` file location and name, you can set it manually.
```

File: vizro-ai/docs/pages/user_guides/api_setup.md [hunk 8]
```diff
@@ -18,16 +18,33 @@ file. Then, you can load the API key from the `.env` file in your development en
...
+The default import of the .env file can be overridden by specifying the path and name.
```
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user_guides/api_setup.md:31]:
```suggestion
The default import of the `.env` file can be overridden by specifying the path and name.
```

**Review by maxschulz-COL — APPROVED** [review]
LGTM! Just a couple of stylistic suggestions

File: vizro-ai/docs/pages/tutorials/explore_vizro_ai.md [hunk 9]
```diff
@@ -13,12 +13,11 @@ Before proceeding, ensure the installation of the `vizro_ai` package by followin
...
-Upon successful setup, load your API key with the following two lines:
+Upon successful setup, your API key should be ready in the environment when you import `vizro_ai`
 
-```py
-from dotenv import load_dotenv
-load_dotenv()
-```
+If you would like to customize the `.env` file location and name, you can set it manually.
+You can override the default import of the `.env` file by specifying the path and name of your custom `.env` file.
+Please refer to [api setup](../user_guides/api_setup.md) for instructions on customizing the `.env` file location and name.
```
Comment (antonymilne) [inline vizro-ai/docs/pages/tutorials/explore_vizro_ai.md:20]:
```suggestion
Please refer to [API setup](../user_guides/api_setup.md) for instructions on customizing the `.env` file location and name.
```

File: vizro-ai/docs/pages/user_guides/api_setup.md [hunk 10]
```diff
@@ -18,16 +18,33 @@ file. Then, you can load the API key from the `.env` file in your development en
...
+#### Customizing the .env file location and name
+If you would like to customize the `.env` file location and name, you can set it manually.
+The default import of the `.env` file can be overridden by specifying the path and name.
+Here's how you can do it:
+```py
+from dotenv import load_dotenv, find_dotenv
+from os.path import join, dirname
+
+# Specify the exact path to your .env file
+env_file = join(dirname(__file__), '.env')  # Adjust the path as needed
```
Comment (antonymilne) [inline vizro-ai/docs/pages/user_guides/api_setup.md:38]:
```suggestion
env_file = Path.cwd() / ".env"  # Adjust the path as needed
```

Let's always try to use `pathlib` for this sort of thing 🙂

(I'm not sure if I mentioned this before but it's a good general piece of advice, e.g. I changed to using pathlib when rewriting your handling of the assets folder in https://github.com/mckinsey/vizro/pull/151/files#diff-cbe161738a476b143cf6226f6f745cd66daef1634fa2e18ded8febd9c91abfde)

File: vizro-ai/docs/pages/user_guides/api_setup.md [hunk 11]
```diff
@@ -18,16 +18,33 @@ file. Then, you can load the API key from the `.env` file in your development en
...
-        from dotenv import load_dotenv
-        load_dotenv()
-        ```
+#### Customizing the .env file location and name
+If you would like to customize the `.env` file location and name, you can set it manually.
+The default import of the `.env` file can be overridden by specifying the path and name.
+Here's how you can do it:
+```py
+from dotenv import load_dotenv, find_dotenv
+from os.path import join, dirname
```
Comment (antonymilne) [inline vizro-ai/docs/pages/user_guides/api_setup.md:35]:
```suggestion
from pathlib import Path
```

**Review by antonymilne — APPROVED** [review]
